internal/pkg/template/diff/diff.go

```diff
@@ -0,0 +1,27 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+import (
+	"gopkg.in/yaml.v3"
+)
+
+// DiffNode represents a difference between two YAML documents.
```

Comment:
nit:
```suggestion
// DiffNode holds the contents of two YAML documents under a particular key.
```

Reply:
out of curiosity: Interesting I wonder how this is going to behave if a key exists in one document but not the other? I am guessing one of `oldValue` or `newValue` will be nil?

Reply:
> I am guessing one of oldValue or newValue will be nil?

This is correct 👍🏼 

> // DiffNode holds the contents of two YAML documents under a particular key.

This is not super duper accurate though - I want to clarify a little to make sure we are on the same page. A node is added to the diff tree iff somewhere under the key, there is a diff:

### Example 1
```
# document 1
Earth:
  Asia: "yo"
  NorthAmerica: "hey"

# document 2
Earth:
  Asia: "yo
  NorthAmerica: "hey"
```
This should yield an empty diff tree with only the sentinel node `&diffNode{}`, because there is no diff

### Example 1
```
# document 1
Earth:
  Asia: "yo"
  NorthAmerica: "hey"

# document 2
Earth:
  NorthAmerica: "hey"
```
This should yield a tree like this:
![Untitled-2023-03-03-1011](https://user-images.githubusercontent.com/79273084/222796660-64171b00-873b-452b-9e6f-005bb4101333.png)

`NorthAmerica` is not in the tree, because there is no diff.